### PR TITLE
Fix routing to account for rawToContentPos

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixup of #16477 

### Summary of the issue:
When writing #16477, I hadn't realized that `TextInfoRegion` compensated for characters that aren't part of the text info. This mapping is stored in the `_rawToContentPos` attribute. This compensation is necessary for character movement based routing, but not for code point offset based routing. The following would happen without this pr.

1. In Word, write a list item, e.g. `1. Hello`
2. Route to the fourth cell (offset 3)
3. The third raw position (h) would correspond with content position zero (1)
4. With `moveToCodepointOffset`, moving to offset 0 means moving to `1` not to `h`. Basically, the cursor would end up at the first cell, not at the fourth.

### Description of user facing changes
Routing works again in Word list items.

### Description of development approach
1. When using `moveToCodepointOffset`, move to the raw position, not to the actual content position. `moveToCodepointOffset` expects raw positions, not content positions.
2. Don't use `moveToCodepointOffset` when the content position is 0, since we can simply collapse the textInfo at the start of the reading unit and return that. Note that `moveToCodepointOffset` would fail for everything up to offset 3 anyway, since it wouldn't be able to move.

### Testing strategy:
Tested the steps to reproduce as mentioned above.

### Known issues with pull request:
None known

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Review by seanbudd — APPROVED** [review]

File: source/braille.py [hunk 1]
```diff
@@ -1345,18 +1345,27 @@ def update(self):
 			self._brailleInputIndStart = None
 
 	def getTextInfoForBraillePos(self, braillePos):
```
[the remark this answers is not in the document]
Comment (LeonarddeR) [reply source/braille.py]: Done

**Comment by burmancomp** [comment]
I have noticed this behavior change:

When selecting text with shift+down arrow in wordpad atleast in windows 10, review position moves to next line after last selected line (to first unselected line).

This does not happen in notepad or in word 2019.

Could this change have been introduced with this pr?

**Comment by LeonarddeR** [comment]
No. I think it's more likely a cause of the behavioral changes you introduced with the several prs of redundant handleUpdate calls.

**Comment by burmancomp** [comment]
They were reverted. It seems to be #16455.

**Comment by burmancomp** [comment]
Unfortunately in browse mode routing buttons seem to work inaccurately at least with firefox in windows 10:

- go to <https://www.nvaccess.org>
- move to end of page (control+end)
- press up arrow to move to "careers"
- press various routing buttons.

**Comment by LeonarddeR** [comment]
Ugh, good catch. It looks like fixing problem 1 introduced problem 2. I think it might be better to revert at least this pr, may be also #16477, I'm not sure. What do you think @seanbudd

**Pull request merged**